app/api/routers/openneuro.py

```diff
@@ -1,105 +1,122 @@
+from github import Auth, GithubIntegration
+from github.GithubException import GithubException, UnknownObjectException
 
 from .. import utility as utils
 from ..dictionary_utils import validate_data_dict
 from ..models import (
+    Contributor,
     FailedUpload,
     SuccessfulUpload,
     SuccessfulUploadWithWarnings,
 )
 
-# TODO: Error out when these variables are not set
-GH_PAT = os.environ.get("GH_PAT")
-API_USERNAME = bytes(os.environ.get("API_USERNAME"), encoding="utf-8")
-API_PASSWORD = bytes(os.environ.get("API_PASSWORD"), encoding="utf-8")
+# TODO: Error out when these variables are not set?
+APP_ID = os.environ.get("NB_BOT_ID")
+APP_PRIVATE_KEY_PATH = os.environ.get("NB_BOT_KEY_PATH")
 
-router = APIRouter(prefix="/openneuro", tags=["openneuro"])
-
-security = HTTPBasic()
-
-
-def verify_credentials(credentials: HTTPBasicCredentials = Depends(security)):
-    """Check if the provided credentials are valid. If not, raise an HTTPException."""
-    current_username_bytes = credentials.username.encode("utf-8")
-    is_correct_username = secrets.compare_digest(
-        current_username_bytes, API_USERNAME
-    )
+DATASETS_ORG = "OpenNeuroDatasets-JSONLD"
 
-    current_password_bytes = credentials.password.encode("utf-8")
-    is_correct_password = secrets.compare_digest(
-        current_password_bytes, API_PASSWORD
-    )
-
-    if not (is_correct_username and is_correct_password):
-        raise HTTPException(
-            status_code=status.HTTP_401_UNAUTHORIZED,
-            detail="Incorrect username or password",
-            headers={"WWW-Authenticate": "Basic"},
-        )
+router = APIRouter(prefix="/openneuro", tags=["openneuro"])
 
 
 # TODO: Factor out main logic into a CRUD function for easier mocking in tests
-# For context on how we use dependencies here, see https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/
 @router.put(
     "/upload",
     response_model=Union[SuccessfulUpload, SuccessfulUploadWithWarnings],
     responses={400: {"model": FailedUpload}},
-    dependencies=[Depends(verify_credentials)],
 )
-async def upload(dataset_id: str, data_dictionary: Annotated[dict, Body()]):
-    # TODO: Handle network errors
-    gh_repo_url = f"https://github.com/OpenNeuroDatasets-JSONLD/{dataset_id}"
-    repo_url = (
-        f"https://api.github.com/repos/OpenNeuroDatasets-JSONLD/{dataset_id}"
+async def upload(
+    dataset_id: str,
+    data_dictionary: Annotated[UploadFile, File()],
+    changes_summary: Annotated[str, Form()],
+    name: Annotated[str, Form()],
+    email: Annotated[str, Form()],
+    affiliation: Annotated[str | None, Form()] = None,
+    # TODO: Do we need to check that it doesn't already start with a "@"? Or should we let the UI handle that?
+    gh_username: Annotated[str | None, Form()] = None,
+):
+    # TODO: Switch to using this Pydantic model directly for the /upload route form data once we
```

Comment:
We’re using Pydantic v1 in this codebase because it relies on data dictionary model/validation modules copied over from the CLI, which were written in v1.